substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -970,7 +971,8 @@ impl<T: Config> Pallet<T> {
         }
 
         Self::_update_contract_state(&mut contract, &types::ContractState::Deleted(cause))?;
-        Self::bill_contract(contract.contract_id)?;
+        let index = Self::get_current_billing_loop_index();
```

Comment:
this is the only thing that "annoys" me...  
because passing no matter what index here will continue to work fine

Reply:
What do you mean? In case the index doens't exist?

Reply:
I guess what Erwan is saying is that when you pass a wrong index but a correct contractID, the contract will be billed normally but not cleaned up if the index passed is wrong.

It's a side-effect of having the code structured like this.